src/v/storage/disk_log_impl.cc

```diff
@@ -568,6 +568,10 @@ ss::future<> disk_log_impl::new_segment(
                 }
                 _segs.add(std::move(h));
                 _probe.segment_created();
+                if (_stm_manager) {
```

Comment:
for the batch cache which is an optional feature we add a helper in the header that hides the `if (_stm_manager)` conditional so the code reads as `make_snapshot()`

Reply:
I've get rid of set_stm_manager so _stm_manager is always there and there is no need to check